File: programmers/기초/2의 영역.js
//2의 영역
function solution(arr) {
  const first = arr.findIndex((a) => a === 2);
  if (first === -1) {
    return [-1];
  }
  console.log(first);
  const last = arr.lastIndexOf(2);
  if (!last) {
    return [2];
  }

  return arr.slice(first, last + 1);
}

// 간단히
function solution2(arr) {
  const from = arr.indexOf(2);
  const end = arr.lastIndexOf(2);

  return from === -1 ? [-1] : arr.slice(from, end + 1);
}

File: programmers/python/2의 영역.py
# 2의 영역
def solution(arr):
    if 2 not in arr:
        return [-1]
    start = arr.index(2)
    end = start
    for i in range(start + 1, len(arr)):
        if arr[i] == 2:
            end = i
    return arr[start:end + 1]